extensions/flush-dns/src/index.ts

```diff
@@ -1,66 +1,134 @@
 import { confirmAlert, showHUD } from "@raycast/api";
-import { SpawnSyncReturns, execSync } from "node:child_process";
+import { execSync } from "node:child_process";
+import { platform } from "node:os";
 
-const isSpawnReturn = (e: unknown): e is SpawnSyncReturns<Buffer> => {
-  return typeof e === "object" && e !== null && "status" in e && "stdout" in e && "stderr" in e;
-};
+type ShellType = string | undefined;
+
+interface CommandSet {
+  commands: Record<string, string>;
+  shell: ShellType;
+}
 
-const commands = {
-  mDNSResponder: "sudo /usr/bin/killall -HUP mDNSResponder",
-  dscacheutil: "sudo /usr/bin/dscacheutil -flushcache",
-  mdnsflushcache: "sudo /usr/bin/discoveryutil mdnsflushcache",
-} as const satisfies Record<string, string>;
+const OS_COMMANDS: Record<string, CommandSet> = {
+  darwin: {
+    commands: {
+      dscacheutil: "/usr/bin/dscacheutil -flushcache",
+      mDNSResponder: "/usr/bin/killall -HUP mDNSResponder",
+      mdnsflushcache: "/usr/bin/discoveryutil mdnsflushcache",
+    },
+    shell: "/bin/bash",
+  },
+  win32: {
+    commands: {
+      ipconfig: "ipconfig /flushdns",
+    },
+    shell: undefined, // Uses cmd.exe by default
+  },
+};
```

Comment:
Since this is not dynamic, I think we should keep it as const for better type hinting.

```suggestion
const OS_COMMANDS = {
  darwin: {
    commands: {
      dscacheutil: "/usr/bin/dscacheutil -flushcache",
      mDNSResponder: "/usr/bin/killall -HUP mDNSResponder",
      mdnsflushcache: "/usr/bin/discoveryutil mdnsflushcache",
    },
    shell: "/bin/bash",
  },
  win32: {
    commands: {
      ipconfig: "ipconfig /flushdns",
    },
    shell: undefined, // Uses cmd.exe by default
  },
} as const satisfies Record<string, CommandSet>;
```

Reply:
The key in Record should probably be partial of the return value of platform() and not string, for good type hinting about what platforms are valid values.